### ISSUE TITLE
No data push in XiaomiGateway device updates

### ISSUE THREAD

**Issue by Masurov** [domoticz/domoticz#3379]
Unlike other device types, XiaomiGateway **Temp only** devices does not trigger data push mechanism.

I looked at the XiaomiGateway.cpp code ~~and it looks like **m_mainworker.sOnDeviceReceived is not called** in InsertUpdateX methods. Other hardware implementations (PythonObject, EvoHome, ...) make this call to trigger data push.~~

Domoticz Stable Version: 4.10717

**Comment by gizmocuz** [comment]
@Masurov , Looking at the code, this should work for (for example) temperature sensors as they are correctly updated in the system by calling the function SendTempSensor
In the code i see some calls to m_mainworker.PushAndWaitRxMessage, this should not be done !
We have functions (like the temp sensor) to send/update switches

It is probably for the switches you are having issues with ?

It should call 'SendGeneralSwitch' or sDecodeRXMessage

**Comment by Masurov** [comment]
@gizmocuz Sorry for the wrong assumption in code.

I did some more testing and it's **working if you use the Temp + Humidity** device in data push link. The device showing temp alone is not working.

I think we can keep the issue open. Best would be to have the temp device working in data push link. Or at least, not showing this type in the device dropdown cause it's not working as is. 

I modified the title to reflect the real issue.

I normally don't use mixed sensor device types in my setup because I find it a bit awkward.
